src/rpc/net.cpp

```diff
@@ -299,10 +299,10 @@ UniValue getaddednodeinfo(const UniValue& params, bool fHelp)
 
     std::vector<AddedNodeInfo> vInfo = GetAddedNodeInfo();
 
-    if (params.size() == 2) {
+    if (params.size() == 2 || params.size() == 1 && params[0].type() == UniValue::VSTR) {
```

Comment:
One downside I just realized: this likely does not work in bitcoin-cli
because it will try to interpret the first argument as JSON.

We could remove the parsing, though, which would make old-style calls fail.